src/lib/y2storage/proposal/space_maker.rb

```diff
@@ -293,39 +374,38 @@ def candidate_disk_names
         settings.candidate_devices
       end
 
-      # Try to resize the existing windows partitions
+      # Next partition to be deleted by {#resize_and_delete}
       #
-      # @param planned_partitions [Array<Planned::Partition>] list of
-      #   partitions to allocate, used to know how much space is still missing
-      # @param disk [String] optional disk name to restrict operations to
-      # @param force [Boolean] whether to resize Windows even if there are
-      #   Linux partitions in the same disk
-      def resize_windows!(planned_partitions, disk, force: false)
-        return unless settings.resize_windows
-        part_names = windows_part_names(disk)
-        return if part_names.empty?
-
-        log.info("Resizing Windows partitions (force: #{force})")
-        parts_by_disk = partitions_by_disk(part_names)
-        remove_linux_disks!(parts_by_disk) unless force
+      # @return [Partition]
+      def next_partition_to_resize
+        log.info("Looking for Windows partitions to resize")
 
-        success = sorted_resizables(parts_by_disk.values.flatten).any? do |part|
-          target_shrink_size = resizing_size(part, planned_partitions, disk)
-          shrink_partition(part, target_shrink_size)
+        return nil unless settings.resize_windows
+        part_names = windows_part_names(disk_name)
+        return nil if part_names.empty?
 
-          success?(planned_partitions)
-        end
+        log.info("Evaluating the following Windows partitions: #{part_names}")
+
+        parts_by_disk = partitions_by_disk(part_names)
+        remove_linux_disks!(parts_by_disk) unless no_more_linux_or_other?
+
+        sorted_resizables(parts_by_disk.values.flatten).first
+      end
 
-        log.info "Didn't manage to free enough space by resizing Windows" unless success
+      # @see #next_partition_to_resize
+      #
+      # @return [Boolean]
+      def no_more_linux_or_other?
```

Comment:
I said above, I would define it as:
```suggestion
      def more_linux_or_other?
        (sorted_candidates[:linux] + sorted_candidates[:other] - new_graph_deleted_sids).any?
      end
      
```